api_docgen/Format_tutorial.mld

```diff
@@ -256,6 +270,23 @@ this:
 )
 ]}
 
+Writing the necessary break hints and boxes in the correct place can be
+error-prone, but we can write some helper functions:
+
+{[
+let structural l r pp ppf v = Format.fprintf ppf "@[%s@[%a@]@,%s@]" l pp v r
+let sparens ppf = structural "(" ")" ppf
+let pair pp1 pp2 ppf (a, b) = Format.fprintf ppf "%a%a" pp1 a pp2 b
+let () =
+  Format.printf "%a"
+    (Format.pp_print_string |> sparens
+    |> pair Format.pp_print_string
+    |> sparens
+    |> pair Format.pp_print_string
+    |> sparens)
+    ("---", ("----", "---"))
+]}
+
```

Comment:
This is out-of-scope of this tutorial, and I don't think introducing combinators to use them once is simpler for the target audience of this tutorial.

Also personally, I find the use of the left-to-right pipeline surprising in this context and  I would propose you to write
```ocaml
let string s ppf = Format.pp_print_string ppf s
let parens pp = Format.dprintf "@[(@[%t@]@,)@]" pp
let pair x y ppf = x ppf; y ppf

let () =
   Format.printf "%t" @@
  parens @@
    pair
      (string "---")
      (parens
         (pair
            (string "----")
            (parens @@ string "---")
         )
      )
```

Reply:
I'll remove it when I update the PR.